diff --git a/internal/p10k.zsh b/internal/p10k.zsh
index 5455723..6a30647 100755
--- a/internal/p10k.zsh
+++ b/internal/p10k.zsh
@@ -892,11 +892,19 @@ prompt_dir() {
       esac
     fi
 
+    local _foreground="$DEFAULT_COLOR"
+    local _background="blue"
+
+    [ -n "$POWERLEVEL9K_DIR_PATH_FOREGROUND" ] && \
+        _foreground="$POWERLEVEL9K_DIR_PATH_FOREGROUND"
+    [ -n "$POWERLEVEL9K_DIR_PATH_BACKGROUND" ] && \
+        _background="$POWERLEVEL9K_DIR_PATH_BACKGROUND"
+
     local style=%b
-    _p9k_color blue $0_$state BACKGROUND
+    _p9k_color $_background $0_$state BACKGROUND
     _p9k_background $_P9K_RETVAL
     style+=$_P9K_RETVAL
-    _p9k_color "$DEFAULT_COLOR" "$0_$state" FOREGROUND
+    _p9k_color "$_foreground" "$0_$state" FOREGROUND
     _p9k_foreground $_P9K_RETVAL
     style+=$_P9K_RETVAL
 
@@ -921,7 +929,7 @@ prompt_dir() {
       sep=$_P9K_RETVAL$sep
     fi
 
-    _p9k_cache_set "$0_$state" "$2" blue "$DEFAULT_COLOR" "$icon" 0 "" "${(pj.$sep.)parts}"
+    _p9k_cache_set "$0_$state" "$2" "$_background" "$_foreground" "$icon" 0 "" "${(pj.$sep.)parts}"
   fi
   "$1_prompt_segment" "$_P9K_CACHE_VAL[@]"
 }
